### ISSUE TITLE
Beta Print Flow

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-ux#63]
We've now got a pretty good idea of the "Export to USB" workflow & design (#57) and need a similar workflow & design for the "Print" action. This is blocked on completion of a proof-of-concept print workflow (https://github.com/freedomofpress/securedrop-workstation/issues/267).

# Acceptance Criteria
- Prototype presented to team w/ input solicited & iterated to team consensus
- Prototype updated and published, with screens published to Zeplin
- Ticket updated w/ MoSCoW scoping for Iteration Zero through any Beta stretch goals

**Comment by ninavizz** [comment]
After kicking around the DispVM name options for a while, I decided I'm not comfortable with a VM name that carries a literal allusion (vs a more conceptual allusion). The literal-ness of "Transfer" I believe, has too much potential to cause confusion... which has been my point of hesitation around it. It also speaks more to the Export action, than it does to the Print action.

I wasn't so much that I was in love with sd-mudroom or sd-vestibule, both of which I do appreciate to be too esoteric... but my desire to keep them in the pool, at least, was to help deduce why I leaned towards them.

sd-mezzo does not reference anything specific, and is also not an English word. It's Italian for "half," and the Disp-VM being an intermediary or halfway point in two different tasks, has me liking it a lot. Not at all married to it specifically, but the deductive logic that had my brain arriving there, I am more married to.

**Comment by eloquence** [comment]
Naming under discussion here:
https://github.com/freedomofpress/securedrop-workstation/issues/285

**Comment by ninavizz** [comment]
_...below, per discussion yesterday with @eloquence where he clarified/reminded me CUPS has been decided to be the spooling thingbob we'll be using for the Workstation._

## CUPS Things

> CUPS is the standards-based, open source printing system developed by Apple Inc. for macOS® and other UNIX®-like operating systems. CUPS uses the Internet Printing Protocol (IPP) to support printing to local and network printers.

- CUPS is the Linux print thingbob (more than a spool, or just a spool?) Qubes will use with Ubuntu/Debian. The above summary was cut-and-pasted from their website.
- [Design Spec](https://www.cups.org/doc/spec-design.html)
![image](https://user-images.githubusercontent.com/8262612/61325700-21929e00-a7ca-11e9-8bda-054e3e370745.png)

**Comment by ninavizz** [comment]
### 15 July: Notes from meeting btwn Nina/Erik
_Note: Most of the below was redundantly cited from Erik to Nina, as details Nina too easily forgot or failed to note in mega info-dump in Mickael's PoC demo_
- Printer needs to be turned on to be detectable
- Yes, ports will still need to be visually marked on the laptop, as sec-arch still needs to designate specific ports as safe for printers and drives
- CUPS is the spooling thingbob
- Discussion about handling printer-origin errors
  - Erik of initial impression it'd be good to articulate specific errors in Client UI, but Nina flagged LCDs on printers typically communicating same things, and mutual agreement made that it could simplify things to instead just flag "Printer side error" in client.

**Comment by ninavizz** [comment]
Tracked work in above-linked Client repo issues, closing as task was long ago completed.
